### ISSUE TITLE
[Bug] Rule I0924 expression-can-be-simplified, is raised when its not valid example ${VAR}==0

### ISSUE THREAD

**Issue by IlfirinPL** [MarketSquare/robotframework-robocop#1174]
### What happened?

Below Code where comparing ${VAR} to 0 is raised as it could be simplified. But when "simplified" will work incorrectly. 
I personally use it for cases when I check return code from process ${result.rc} 

### What command/code did you try to run?

    VAR    ${VAR}    0
    Should Be True    ${VAR}==0    Error 1
    Should Be True    ${VAR}    Error 2

### What is the full error message?

Error 2

### What did you expect to happen instead?

I expect to not raise rule violation  I0924 for ${VAR}==0

### Operating System

N/A

### Robocop version

5.8.1

**Comment by bhirsz** [comment]
I believe we have similiar issue in the backlog complaning about this rule - I need to make it as non-default because of such false positives. 

In your example you can always do ``$VAR == 0`` which probably will not trigger warning, but of course your code is also fine.

**Comment by IlfirinPL** [comment]
In code also raise issue when "space" is added
  IF    ${result.rc} == 0

![Image](https://github.com/user-attachments/assets/cb5a1ac0-a04c-477a-b66b-315f25623668)

**Comment by IlfirinPL** [comment]
I will provide another example "expression-can-be-simplified", raise issue when its not valid

IF    ${connected} == ${False}

**Comment by bhirsz** [comment]
I have removed 0 from condition of this rule as it was too often false positive. I also made rule non-default (opt-in). 

One question:
```
IF ${connected} == ${False}
```
why following simplified condition is not valid?
```
IF    not $connected
```

**Comment by IlfirinPL** [comment]
Its good question, this part should be auto transform :)
Since 
`IF ${connected} == ${False}` 
and
`IF ${connected} == ${EMPTY}`
will give the same results
as
`IF    not $connected`

only against is that using ${VAR}  syntax will rise issue 
`IF    not ${connected}` 
will rise error 
Invalid IF condition: Evaluating expression 'not ' failed: SyntaxError: invalid syntax (<string>, line 1)
